routes/mapper.py

```diff
@@ -162,25 +162,25 @@ def __init__(self, obj, resource_name=None, collection_name=None,
                 self.formatted = True
         self.add_actions(actions or [], **kwargs)
 
-    def connect(self, *args, **kwargs):
+    def connect(self, routename, path=None, **kwargs):
         newkargs = {}
-        # newargs = args
-        routename, path = args
+        _routename = routename
+        _path = path
         for key, value in six.iteritems(self.kwargs):
             if key == 'path_prefix':
-                if len(args) > 1:
+                if path is not None:
                     # if there's a name_prefix, add it to the route name
                     # and if there's a path_prefix
-                    path = ''.join((self.kwargs[key], args[1]))
+                    _path = ''.join((self.kwargs[key], path))
```

Comment:
why iterating on key+value if later you retrieve the value from self.kwargs[key]? => use value, no?